### ISSUE TITLE
Async-Input: Throw error if missing name attribute

### ISSUE THREAD

**Issue by torch2424** [ampproject/amphtml#20338]
From some early feedback from the Google Pay team on building their extension, they hit a roadblock with trying to understand the name attribute. amp-form should look for this attribute, and assert it, so that the error can help developers with the misunderstanding going forward.

**Comment by lannka** [comment]
Not sure if this is still needed.
